research-app/src/SourceItem.vue

```diff
@@ -0,0 +1,224 @@
+      <div v-if="isSelected" class="detail-container">
+        <div class="detail-row"><span class="prompt">RA:</span><span>{{ this.raStr }}</span></div>
+        <div class="detail-row"><span class="prompt">Dec:</span><span>{{ this.decStr }}</span></div>
+        <div class="detail-row"><span class="prompt">Catalog:</span><span>{{ this.source.catalogName }}</span></div>
+        <div class="detail-row"><span class="prompt">Query Coordinates:</span><a :href="this.simbadCoordinatesURL" target="_blank" rel="noopener noreferrer">SIMBAD</a> | 
+        <a :href="this.nedCoordinatesURL" target="_blank" rel="noopener noreferrer">NED</a></div>
+      </div>
+    </transition-expand>
+  </div>
+</template>
+
+<script lang="ts">
+import { mapActions, mapMutations, mapState } from "vuex";
+import { Component, Prop, Vue, Watch } from "vue-property-decorator";
+
+import { Source } from "./store";
+import { wwtEngineNamespace, wwtResearchAppNamespace } from "./namespaces";
+import { WWTEngineVuexState } from "@wwtelescope/engine-vuex";
+import { fmtDegLat, fmtHours } from "@wwtelescope/astro";
+
+const R2D = 180 / Math.PI;
+
+@Component
+export default class SourceItem extends Vue {
+    @Prop({ required:true }) source!: Source;
+    hasFocus = false;
+    isSelected = false;
+    editing = false;
+
+    // From the stores
+    wwtDegZoom!: number;
+    gotoRADecZoom!: (args: { raRad: number; decRad: number; zoomDeg: number; instant: boolean; rollRad?: number }) => Promise<void>;
+    removeSource!: (source: Source) => void;
+
+    beforeCreate(): void {
+      this.$options.computed = {
+        ...mapState(wwtEngineNamespace, {
+            wwtDegZoom: (state, _getters) => (state as WWTEngineVuexState).zoomDeg,
+        }),
+        ...this.$options.computed,
+      }
+
+      this.$options.methods = {
+          ...this.$options.methods,
+          ...mapActions(wwtEngineNamespace, [
+            "gotoRADecZoom"
+          ]),
+          ...mapMutations(wwtResearchAppNamespace, [
+            "removeSource",
+            "setSourceName",
+          ])
+      }
+    }
+
+    handleDelete() {
+      this.removeSource(this.source);
+    }
+
+    handleMarkerClick() {
+      this.gotoRADecZoom({ zoomDeg: this.source.zoomDeg ?? this.wwtDegZoom, raRad: this.source.ra, decRad: this.source.dec, instant: false }).catch(err => console.log(err));
+    }
+
+
+    handleEditClick() {
+      this.editing = !this.editing;
+    }
+
+    @Watch('editing')
+    editingChanged(val: boolean, _oldVal: boolean) {
+      const input: HTMLInputElement = this.$el.getElementsByClassName('name-input')[0] as HTMLInputElement;
+      if (val) {
+        input.value = this.sourceName;
+      } else {
+        this.sourceName = input.value;
+      }
+    }
+
+    get raStr() {
+      return fmtHours(this.source.ra);
+    }
+
+    get decStr() {
+      return fmtDegLat(this.source.dec);
+    }
+
+    get searchRadius() {
+      return 2 * this.wwtDegZoom / 360; // In arcminutes
+    }
```

Comment:
This should be clamped to never go below a certain minimum value, say 10 arcseconds. When I was playing around I got a case where a SIMBAD search failed because the search radius was like sub-arcsecond, and in realistic use cases the online databases aren't going to necessarily agree with the catalogs to that level precision.

Reply:
Makes sense. I've set 10 arcseconds (= 1/6 arcminutes) as the minimum value.